### PR TITLE
Fix BulkDeleteError showing incorrect values

### PR DESCRIPTION
**Pull request by davfsa**
### Summary
Funnily enough, this also speeds it up a bit :P

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/rest.py [hunk 1]
```diff
@@ -1564,8 +1563,8 @@ async def delete_messages(
     ) -> None:
         route = routes.POST_DELETE_CHANNEL_MESSAGES_BULK.compile(channel=channel)
 
-        pending: typing.Deque[snowflakes.SnowflakeishOr[messages_.PartialMessage]] = collections.deque()
-        deleted: typing.Deque[snowflakes.SnowflakeishOr[messages_.PartialMessage]] = collections.deque()
+        pending = []
+        deleted = []
```
Comment (davfsa) [inline hikari/impl/rest.py:1567]:
```suggestion
        pending: typing.List[snowflakes.SnowflakeishOr[messages_.PartialMessage]] = []
        deleted: typing.List[snowflakes.SnowflakeishOr[messages_.PartialMessage]] = []
```

**Comment by FasterSpeeding** [comment]
bugfix changelog entry?

**Comment by davfsa** [comment]
> bugfix changelog entry?

Added :+1:

File: hikari/impl/rest.py [hunk 2]
```diff
@@ -1593,15 +1592,18 @@ async def delete_messages(
             # I am just gonna invoke these sequentially instead.
             try:
                 if len(pending) == 1:
-                    message = pending.popleft()
+                    message = pending[0]
```
Comment (FasterSpeeding) [inline hikari/impl/rest.py:1595]: can't this still just be `pop(0)`

File: hikari/impl/rest.py [hunk 3]
```diff
@@ -1593,15 +1592,18 @@ async def delete_messages(
...
                     await self.delete_message(channel, message)
                     deleted.append(message)
+
                 else:
                     body = data_binding.JSONObjectBuilder()
-                    chunk = [pending.popleft() for _ in range(min(100, len(pending)))]
+                    chunk = pending[: min(100, len(pending))]
```
Comment (FasterSpeeding) [inline hikari/impl/rest.py:1601]: This is min statement unnecessary, `list[:100]` will give you up to 100 entries but less if there's not 100 entries present
e.g.
```py
>>> [1,2,3,4][:100]
[1, 2, 3, 4]
```

File: hikari/impl/rest.py (hunk 2 above)
Comment (FasterSpeeding) [reply hikari/impl/rest.py:1595]: hmm well you can't really be sure that the message wasn't deleted before the error was raised/returned but sure

Comment (FasterSpeeding) [reply hikari/impl/rest.py:1595]: also if the error was NotFoundError cause this was deleting a single message it wasn't a failed delete

**Pull request merged**